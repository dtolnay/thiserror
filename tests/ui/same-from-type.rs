use thiserror::Error;

#[derive(Error, Debug)]
pub enum Error {
    #[error("failed to open")]
    OpenFile(#[from] std::io::Error),
    #[error("failed to close")]
    CloseFile(#[from] std::io::Error),
}

fn main() {}
