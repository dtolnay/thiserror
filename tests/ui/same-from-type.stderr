error[E0119]: conflicting implementations of trait `From<std::io::Error>` for type `Error`
 --> tests/ui/same-from-type.rs:8:15
  |
6 |     OpenFile(#[from] std::io::Error),
  |              ------- first implementation here
7 |     #[error("failed to close")]
8 |     CloseFile(#[from] std::io::Error),
  |               ^^^^^^^ conflicting implementation for `Error`
